index.js

```diff
@@ -41,6 +43,13 @@ module.exports = function block (size, opts) {
       var target = []
       var b, end, out
 
+      if (buffered.length > SQUASH_BUFFER_THRESHOLD) {
+        buffered = [
+          Buffer.concat(buffered.slice(0, buffered.length - 1)),
+          buffered[buffered.length - 1]
```

Comment:
why do you need to make this two pieces, and not all into one buffer?

Reply:
Hm, a line of thought that carried over from my perfect solution attempts. It was intended as preventing copies for data that wouldn't be part of the output block. But given the larger threshold  and the unlikeliness this will save many bytes it might as well be one buffer.